### PR TITLE
candy: add package

### PR DESCRIPTION
**Pull request by lanthora**
Another VPN that supports peer-to-peer connections.

### CODE REVIEW

**Comment by auxiliarypower** [comment]
1. Please add a PKGBUILD file to the repository.
2. The compilation error is as follows:
```
CMake Error at /usr/share/cmake/Modules/ExternalProject.cmake:2911 (message):
  error: could not find git for clone of ixwebsocket-populate
Call Stack (most recent call first):
  /usr/share/cmake/Modules/ExternalProject.cmake:4422 (_ep_add_download_command)
  CMakeLists.txt:29 (ExternalProject_Add)


-- Configuring incomplete, errors occurred!

CMake Error at /usr/share/cmake/Modules/FetchContent.cmake:1667 (message):
  CMake step for ixwebsocket failed: 1
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FetchContent.cmake:1819:EVAL:2 (__FetchContent_directPopulate)
  /usr/share/cmake/Modules/FetchContent.cmake:1819 (cmake_language)
  cmake/ixwebsocket.cmake:14 (FetchContent_Populate)
  CMakeLists.txt:25 (include)


-- Configuring incomplete, errors occurred!
==> ERROR: A failure occurred in build().
    Aborting...
```

**Comment by lanthora** [comment]
> 1. Please add a PKGBUILD file to the repository.

Done

> 2. The compilation error is as follows:

Add _git_ as _makedepends_ based on [AUR](https://aur.archlinux.org/packages/candy) version

File: archlinuxcn/candy/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+pkgver=3.7
+pkgrel=1
+pkgdesc="A easy-to-deploy and peer-to-peer virtual private network"
+url="https://github.com/lanthora/candy"
+license=('MIT')
+arch=('x86_64' 'aarch64' 'armv7h' 'riscv64')
+source=("https://github.com/lanthora/candy/archive/refs/tags/v$pkgver.tar.gz" )
+sha256sums=('a2cbfdcf2350ee41bfe388a9140c3cf5ef76da592ee4ae5ac20a7841fd38dc16')
+makedepends=('cmake' 'make' 'pkgconf' 'gcc' 'git')
+depends=('spdlog' 'openssl' 'libconfig' 'uriparser')
```
Comment (auxiliarypower) [inline archlinuxcn/candy/PKGBUILD:11]:
```
candy E: Dependency zlib detected and not included (libraries ['usr/lib/libz.so.1'] needed in files ['usr/bin/candy'])
candy W: Dependency fmt detected and implicitly satisfied (libraries ['usr/lib/libfmt.so.10'] needed in files ['usr/bin/candy'])
candy W: Dependency glibc detected and implicitly satisfied (libraries ['usr/lib/libm.so.6', 'usr/lib/libc.so.6', 'usr/lib/ld-linux-x86-64.so.2'] needed in files ['usr/bin/candy'])
candy W: Dependency gcc-libs detected and implicitly satisfied (libraries ['usr/lib/libstdc++.so.6', 'usr/lib/libgcc_s.so.1'] needed in files ['usr/bin/candy'])
candy W: Dependency included, but may not be needed ('spdlog')
```

File: archlinuxcn/candy/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+build() {
+        cd "$pkgname-$pkgver/build"
+        cmake -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_BUILD_TYPE=Release ..
+        make
+}
+
+package() {
+        cd "$pkgname-$pkgver"
+        install -D -m 600 candy.conf "$pkgdir/etc/candy.conf"
```
Comment (auxiliarypower) [inline archlinuxcn/candy/PKGBUILD:22]:
```
candy W: File (etc/candy.conf) does not have the world readable bit set.
```

File: archlinuxcn/candy/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+depends=('spdlog' 'openssl' 'libconfig' 'uriparser')
+backup=('etc/candy.conf')
+
+build() {
+        cd "$pkgname-$pkgver/build"
+        cmake -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_BUILD_TYPE=Release ..
+        make
+}
+
+package() {
```
Comment (auxiliarypower) [inline archlinuxcn/candy/PKGBUILD:20]:
```
candy E: Missing custom license directory (usr/share/licenses/candy)
```

File: archlinuxcn/candy/PKGBUILD
Comment (auxiliarypower) [inline archlinuxcn/candy/PKGBUILD:1]:
```
PKGBUILD (candy) W: Missing Maintainer tag
PKGBUILD (candy) W: Non-unique source name (v3.7.tar.gz). Use a unique filename.
```

File: archlinuxcn/candy/PKGBUILD (hunk 1 above)
Comment (auxiliarypower) [inline archlinuxcn/candy/PKGBUILD:11]:
```
candy E: Dependency zlib detected and not included (libraries ['usr/lib/libz.so.1'] needed in files ['usr/bin/candy'])
candy W: Dependency fmt detected and implicitly satisfied (libraries ['usr/lib/libfmt.so.10'] needed in files ['usr/bin/candy'])
candy W: Dependency glibc detected and implicitly satisfied (libraries ['usr/lib/libm.so.6', 'usr/lib/libc.so.6', 'usr/lib/ld-linux-x86-64.so.2'] needed in files ['usr/bin/candy'])
candy W: Dependency gcc-libs detected and implicitly satisfied (libraries ['usr/lib/libstdc++.so.6', 'usr/lib/libgcc_s.so.1'] needed in files ['usr/bin/candy'])
candy W: Dependency included, but may not be needed ('spdlog')
```

File: archlinuxcn/candy/PKGBUILD (hunk 3 above)
Comment (auxiliarypower) [inline archlinuxcn/candy/PKGBUILD:20]:
```
candy E: Missing custom license directory (usr/share/licenses/candy)
```

File: archlinuxcn/candy/PKGBUILD (hunk 2 above)
Comment (auxiliarypower) [inline archlinuxcn/candy/PKGBUILD:22]:
```
candy W: File (etc/candy.conf) does not have the world readable bit set.
```

**Comment by auxiliarypower** [comment]
> Add _git_ as _makedepends_ based on [AUR](https://aur.archlinux.org/packages/candy) version

Please also update PKGBUILD file on Arch User Repository.

File: archlinuxcn/candy/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,27 @@
+pkgname=candy
+pkgver=3.7
+pkgrel=1
+pkgdesc="A easy-to-deploy and peer-to-peer virtual private network"
+url="https://github.com/lanthora/candy"
+license=('MIT')
+arch=('x86_64' 'aarch64' 'armv7h' 'riscv64')
+source=("https://github.com/lanthora/candy/archive/refs/tags/v$pkgver.tar.gz" )
+sha256sums=('a2cbfdcf2350ee41bfe388a9140c3cf5ef76da592ee4ae5ac20a7841fd38dc16')
```
Comment (auxiliarypower) [inline archlinuxcn/candy/PKGBUILD:9]:
```
PKGBUILD (candy) W: Non-unique source name (v3.7.tar.gz). Use a unique filename.
```

File: archlinuxcn/candy/PKGBUILD [hunk 5]
```diff
@@ -0,0 +1,27 @@
+pkgname=candy
+pkgver=3.7
```
Comment (auxiliarypower) [inline archlinuxcn/candy/PKGBUILD:2]:
```
PKGBUILD (candy) W: Missing Maintainer tag
```

**Comment by lanthora** [comment]
All issues reported by namcap have been fixed

File: archlinuxcn/candy/lilac.yaml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+maintainers:
+  - github: lanthora
+    email: lanthora@outlook.com
+
+pre_build_script: aur_pre_build(maintainers=['lanthora'])
+post_build: aur_post_build
+
+update_on:
+  - source: aur
```
Comment (auxiliarypower) [inline archlinuxcn/candy/lilac.yaml:9]:
```
update_on:
  - source: aur
    aur: candy
```

**Pull request merged**